packages/google_maps_flutter/google_maps_flutter_android/android/src/test/java/io/flutter/plugins/googlemaps/ConvertTest.java

```diff
@@ -26,4 +72,163 @@ public void ConvertToPointsConvertsThePointsWithFullPrecision() {
+
+    when(flutterInjectorWrapper.getLookupKeyForAsset(fakeAssetName)).thenReturn(fakeAssetKey);
+
+    when(assetManager.open(fakeAssetKey)).thenReturn(buildImageInputStream());
+
+    when(bitmapDescriptorFactoryWrapper.fromBitmap(any())).thenReturn(mockBitmapDescriptor);
+
+    BitmapDescriptor result =
+        Convert.getBitmapFromAsset(
+            data, assetManager, 1.0f, bitmapDescriptorFactoryWrapper, flutterInjectorWrapper);
+
+    Assert.assertEquals(mockBitmapDescriptor, result);
+  }
+
+  @Test
+  public void GetBitmapFromAssetNoScaling() throws Exception {
+    String fakeAssetName = "fake_asset_name";
+    String fakeAssetKey = "fake_asset_key";
+    ArrayList<Object> data =
+        new ArrayList<>(Arrays.asList("asset", fakeAssetName, "noScaling", 2.0f));
+
+    when(flutterInjectorWrapper.getLookupKeyForAsset(fakeAssetName)).thenReturn(fakeAssetKey);
+
+    when(assetManager.open(fakeAssetKey)).thenReturn(buildImageInputStream());
+
+    when(bitmapDescriptorFactoryWrapper.fromAsset(any())).thenReturn(mockBitmapDescriptor);
+
+    verify(bitmapDescriptorFactoryWrapper, never()).fromBitmap(any());
+
+    BitmapDescriptor result =
+        Convert.getBitmapFromAsset(
+            data, assetManager, 1.0f, bitmapDescriptorFactoryWrapper, flutterInjectorWrapper);
+
+    Assert.assertEquals(mockBitmapDescriptor, result);
+  }
+
+  @Test(expected = IllegalArgumentException.class) // Expecting an IllegalArgumentException
+  public void GetBitmapFromAssetThrowsErrorIfAssetNotAvailable() throws Exception {
+    String fakeAssetName = "fake_asset_name";
+    ArrayList<Object> data =
+        new ArrayList<>(Arrays.asList("asset", fakeAssetName, "noScaling", 2.0f));
+
+    when(flutterInjectorWrapper.getLookupKeyForAsset(fakeAssetName))
+        .thenThrow(new RuntimeException("Fake exception"));
+
+    try {
+      Convert.getBitmapFromAsset(
+          data, assetManager, 1.0f, bitmapDescriptorFactoryWrapper, flutterInjectorWrapper);
+    } catch (IllegalArgumentException e) {
+      assertTrue(e.getMessage().startsWith("'asset' cannot open asset: "));
+      throw e; // rethrow the exception
+    }
+
+    fail("Expected an IllegalArgumentException to be thrown");
+  }
+
+  @Test
+  public void GetBitmapFromBytesAuto() throws Exception {
+    String base64Image =
+        "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg==";
+    byte[] bmpData = Base64.decode(base64Image, Base64.DEFAULT);
+
+    List<?> data = new ArrayList<>(Arrays.asList("bytes", bmpData, "auto", 2.0f));
+
+    when(bitmapDescriptorFactoryWrapper.fromBitmap(any())).thenReturn(mockBitmapDescriptor);
+
+    BitmapDescriptor result = Convert.getBitmapFromBytes(data, 1f, bitmapDescriptorFactoryWrapper);
+
+    Assert.assertEquals(mockBitmapDescriptor, result);
+  }
+
+  @Test
+  public void GetBitmapFromBytesAutoAndSize() throws Exception {
+    String base64Image =
```

Comment:
Can you add a comment for how you generated this string and how others can create a new version if they need to later?